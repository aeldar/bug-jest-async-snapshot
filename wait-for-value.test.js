const waitForValue = require('./wait-for-value');

const shortTimeout = 1000;
const longTimeout = 6000;

describe('Jest async experiment', async() => {
    it('Long timeout, more than default 5000', async() => {
        expect.assertions(1);
        const testValue = 'one';
        const result = await waitForValue(longTimeout, testValue);
        expect(result).toMatchSnapshot();
    });
    it('Short timeout', async() => {
        expect.assertions(1);
        const testValue = 'two';
        const result = await waitForValue(shortTimeout, testValue);
        expect(result).toMatchSnapshot();
    });
    it('Short timeout one more time', async() => {
        expect.assertions(1);
        const testValue = 'three';
        const result = await waitForValue(shortTimeout, testValue);
        expect(result).toMatchSnapshot();
    });
});