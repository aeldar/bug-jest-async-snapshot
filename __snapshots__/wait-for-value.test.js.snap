// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest async experiment Long timeout, more than default 5000 1`] = `"one"`;

exports[`Jest async experiment Short timeout 1`] = `"two"`;

exports[`Jest async experiment Short timeout one more time 1`] = `"three"`;
